config/config.go

```diff
@@ -537,6 +539,12 @@ func New(v *viper.Viper) (*Configuration, error) {
 		return nil, err
 	}
 
+	// Migrate global settings to default account
+	if c.AccountRequired == false {
```

Comment:
It is difficult to preserve the original behavior of account_required, as we introduced new cases:
before:
1. pubID missing, account_required=true   -> FAIL
2. pubID missing, account_required=false -> PASS
now, in addition to those:
3. pubID provided, account not found, account_required=true -> I'd like this to FAIL but must be PASS to support the existing behavior (otherwise we'd need to required all the accounts to exist first)
4. pubID provided, account not found, account_required=false -> PASS but use global setting aka default account.

This suggests a single setting might not be sufficient to cover this.

Reply:
I've resolved to adding a new explicit setting `ValidAccountRequired` to control the more strict requirement,
as managing it through a combination of flags was non-intuitive. 

It also makes migration easier.

The existing `AccountRequired`=`true` behavior is preserved. We accept any non-empty pubID and create an account for it from the `AccountDefaults` config, allowing the host to start populating accounts progressively (valid accounts always work). When the host is ready to switch to more strict validation, the `ValidAccountRequired` is turned on at which point pubIDs that don't have an explicit account will be rejected.